capa/main.py

```diff
@@ -533,74 +605,25 @@ def get_extractor(
       UnsupportedOSError
     """
     if format_ not in (FORMAT_SC32, FORMAT_SC64):
-        if not is_supported_format(path):
-            raise UnsupportedFormatError()
-
-        if not is_supported_arch(path):
-            raise UnsupportedArchError()
-
-        if os_ == OS_AUTO and not is_supported_os(path):
-            raise UnsupportedOSError()
+        check_unsupported_raise_exception(path, os_)
```

Comment:
I'd propose to just leave the code verbatim here instead of in a new function. Or do you see much benefit added by the function?